apps/dashboard/src/app/(app)/team/[team_slug]/[project_slug]/(sidebar)/insight/components/TopEndpointsTable.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+import { useMemo } from "react";
+import type { ThirdwebClient } from "thirdweb";
+import { shortenLargeNumber } from "thirdweb/utils";
+import { ScrollShadow } from "@/components/ui/ScrollShadow/ScrollShadow";
+import { SkeletonContainer } from "@/components/ui/skeleton";
+import type { InsightEndpointStats } from "@/types/analytics";
+import {
+  CardHeading,
+  TableData,
+  TableHeading,
+  TableHeadingRow,
+} from "../../universal-bridge/components/common";
+
+const NUMBER_OF_ENDPOINTS_TO_SHOW = 10;
+
+export function TopInsightEndpointsTable(props: {
+  data: InsightEndpointStats[];
+  client: ThirdwebClient;
+}) {
+  const tableData = useMemo(() => {
+    return props.data
+      ?.sort((a, b) => b.totalRequests - a.totalRequests)
+      .slice(0, NUMBER_OF_ENDPOINTS_TO_SHOW);
+  }, [props.data]);
+  const isEmpty = useMemo(() => tableData.length === 0, [tableData]);
+
+  return (
+    <div className="flex flex-col">
+      {/* header */}
+      <div className="flex flex-col gap-2 lg:flex-row lg:items-center lg:justify-between">
+        <CardHeading>Top Endpoints</CardHeading>
+      </div>
+
+      <div className="h-5" />
+      <ScrollShadow
+        className="overflow-hidden rounded-lg border"
+        disableTopShadow={true}
+        scrollableClassName="h-[280px]"
+      >
+        <table className="w-full">
+          <thead>
+            <TableHeadingRow>
```

Comment:
Use shadcn table components here, remove Scrollshadow ( when you'll add - TableContainer, it already includes ScrollShadow )